### PR TITLE
rootd: do not use popen() to expand path names

### PR DESCRIPTION
**Pull request by amadio**
Address security threat reported by S. Luders.
Using the same technology used in TSystem::ExpandFileName.

This is a backport of commit 88ccff152604e0f1012653a596d802ff7ede3145. This is a fix for [CVE-2017-1000203](https://nvd.nist.gov/vuln/detail/CVE-2017-1000203).

### CODE REVIEW

**Comment by gganis** [comment]
Looks good to me. It fails on mac for completely unrelated reasons ...

File: net/rootd/src/rootd.cxx [hunk 1]
```diff
@@ -465,96 +442,198 @@ static const char *HomeDirectory(const char *name)
 }
 
 //______________________________________________________________________________
-char *RootdExpandPathName(const char *name)
+static const char *WorkingDirectory()
```
Comment (pcanal) [inline net/rootd/src/rootd.cxx]: Why not use: TSystem::GetWorkingDirectory ?

File: net/rootd/src/rootd.cxx [hunk 2]
```diff
@@ -465,96 +442,198 @@ static const char *HomeDirectory(const char *name)
...
 {
-   // Expand a pathname getting rid of special shell characters like ~.$, etc.
-   // Returned string must be freed by caller.
+   // Returns the current working directory.
 
-   const char *patbuf = name;
+   static char path[kMAXPATHLEN];
```
Comment (pcanal) [inline net/rootd/src/rootd.cxx]: If really needing a static variable, please make it also thread local.

File: net/rootd/src/rootd.cxx [hunk 3]
```diff
@@ -465,96 +442,198 @@ static const char *HomeDirectory(const char *name)
...
-   for (const char *p = patbuf; *p; p++)
-      if (strchr(shellMeta, *p))
-         goto needshell;
+   if (getcwd(path, kMAXPATHLEN)) return path;
+   return 0;
+}
 
-   return strdup(name);
+//______________________________________________________________________________
+Bool_t RootdExpandFileName(const char *fname, char *xname, const int kBufSize)
```
Comment (pcanal) [inline net/rootd/src/rootd.cxx]: Is this routine local to this file?  Shouldn't it be static and/or unnamed-namespace?

**Comment by pcanal** [comment]
Humm .. some of my comments are only of minor important since I had failed to notice this rootd (standalone app).

**Comment by amadio** [comment]
This is strictly a backport of another commit, and because of that, follows as much as possible what was done there.

**Comment by pcanal** [comment]
fair enough.  Sorry for the noise :)

**Comment by pcanal** [comment]
The build errors are unrelated to this PR.

**Pull request merged**